File: app-module/ClientApp/src/app/announcement.ts
export interface Announcement {
  idAnnouncements: number;
  title: string;
  description: string;
  type: string;
  link: string;
  background: string;
  publishDate: Date;
  expirationDate: Date;
  isDraft: boolean;
}

export function blankAnnouncement(id: number): Announcement{
  return {
      idAnnouncements: id,
      title: "",
      description: "",
      type: "",
      link: "",
      background: "",
      publishDate: new Date(0),
      expirationDate: new Date(0),
      isDraft: false
  }
} 

File: app-module/DataModels/Announcement.cs
using System.ComponentModel.DataAnnotations;

namespace mobile_app_messaging_module.DataModels
{
    public class Announcement
    {
        [Key]
        public int idAnnouncements { get; set; }
        public string? title { get; set; }
        public string? description { get; set; }
        public string? type { get; set; }
        public string? link { get; set; }
        public string? background { get; set; }
        public DateTime? publishDate { get; set; }
        public DateTime? expirationDate { get; set; }
        public Boolean? isDraft { get; set; }
        public int? priority { get; set; }
    }
}
